format/format.go

```diff
@@ -451,6 +474,52 @@ func (w *writer) writeTermParens(parens bool, term *ast.Term, comments []*ast.Co
 	return comments
 }
 
+func (w *writer) writeRef(x ast.Ref) {
+	if len(x) > 0 {
+		w.write(x[0].Value.String())
+		path := x[1:]
+		for _, p := range path {
+			switch p := p.Value.(type) {
+			case ast.String, ast.Var:
+				w.writeRefPath(p)
+			default:
+				w.write("[" + p.String() + "]")
+			}
+		}
+	}
+}
+
+var varRegexp = regexp.MustCompile("^[[:alpha:]_][[:alpha:][:digit:]_]*$")
+
+func (w *writer) writeRefPath(value interface{}) {
+	var str string
+	var formatted string
+	switch v := value.(type) {
+	case ast.String:
+		formatted = v.String()
+		str = string(v)
+	case ast.Var:
+		formatted = w.formatVar(v)
+		str = formatted
+	}
+
+	if varRegexp.MatchString(str) && !ast.IsKeyword(str) && str != ast.Wildcard.String() {
+		w.write("." + str)
```

Comment:
This will produce the wrong answer for refs containing vars. E.g., `input.x[y]` => `input.x.y`.